installer/PowerToysBootstrapper/bootstrapper/bootstrapper.cpp

```diff
@@ -59,72 +65,96 @@ std::optional<fs::path> extractIcon()
-        { L"--no_start_pt", CmdArgs::noStartPT },
-        { L"--skip_dotnet_install", CmdArgs::skipDotnetInstall }
-    };
-}
-
-std::unordered_set<CmdArgs> parseCmdArgs(const int nCmdArgs, LPWSTR* argList)
+void setup_log(const spdlog::level::level_enum severity)
 {
-    std::unordered_set<CmdArgs> result;
-    for (size_t i = 1; i < nCmdArgs; ++i)
+    try
     {
-        if (auto it = knownArgs.find(argList[i]); it != end(knownArgs))
+        auto logger = spdlog::basic_logger_mt("file", LOG_FILENAME);
+        logger->set_pattern("[%L][%d-%m-%C-%T] %v");
+        logger->set_level(severity);
+        spdlog::set_default_logger(std::move(logger));
+        spdlog::set_level(severity);
+        spdlog::flush_every(std::chrono::seconds(5));
+
+        if (severity != spdlog::level::off)
         {
-            result.emplace(it->second);
+            const DWORD msiSev = severity == spdlog::level::debug ? INSTALLLOGMODE_VERBOSE : INSTALLLOGMODE_ERROR;
+            const auto msiLogPath = fs::current_path() / MSI_LOG_FILENAME;
+            MsiEnableLogW(msiSev, msiLogPath.c_str(), INSTALLLOGATTRIBUTES_APPEND);
         }
     }
-    return result;
+    catch (...)
+    {
+    }
 }
 
 int bootstrapper()
 {
     using namespace localized_strings;
     winrt::init_apartment();
-
-    int nCmdArgs = 0;
-    LPWSTR* argList = CommandLineToArgvW(GetCommandLineW(), &nCmdArgs);
-    const auto cmdArgs = parseCmdArgs(nCmdArgs, argList);
-    std::wostringstream oss;
-    if (cmdArgs.contains(CmdArgs::showHelp))
+    cxxopts::Options options{ "PowerToysBootstrapper" };
+    // clang-format off
+    options.add_options()
+        ("h,help", "Show help")
+        ("no_full_ui", "Use reduced UI for MSI")
+        ("s,silent", "Suppress MSI UI and notifications")
+        ("no_start_pt", "Do not launch PowerToys after the installation is complete")
+        ("skip_dotnet_install", "Skip dotnet 3.X installation even if it's not detected")
+        ("log_level", "Log level. Possible values: off|debug|error", cxxopts::value<std::string>()->default_value("off"));
+    // clang-format on
+    cxxopts::ParseResult cmdArgs;
+    options.allow_unrecognised_options();
+    try
     {
-        oss << "Supported arguments:\n\n";
-        for (auto [arg, _] : knownArgs)
-        {
-            oss << arg << '\n';
-        }
-        MessageBoxW(nullptr, oss.str().c_str(), L"Help", MB_OK | MB_ICONINFORMATION);
+        cmdArgs = options.parse(__argc, const_cast<const char**>(__argv));
+    }
+    catch (...)
+    {
+    }
+    const bool showHelp = cmdArgs["help"].as<bool>();
+    const bool noFullUI = cmdArgs["no_full_ui"].as<bool>();
+    const bool silent = cmdArgs["silent"].as<bool>();
+    const bool skipDotnetInstall = cmdArgs["skip_dotnet_install"].as<bool>();
+    const bool noStartPT = cmdArgs["no_start_pt"].as<bool>();
+    const auto logLevel = cmdArgs["log_level"].as<std::string>();
+    spdlog::level::level_enum severity = spdlog::level::off;
```

Comment:
I would move this into the if/else statement to make the code better readable. Maybe a switch/select case with default value makes here sense to save space.

Reply:
I'd rather not do it, because it's important to keep the values default-initialized in C++. if not, it could be a source of subtle bugs in the future.